x/upgrade/keeper/keeper.go

```diff
@@ -228,6 +229,23 @@ func (k Keeper) GetUpgradedConsensusState(ctx sdk.Context, lastHeight int64) ([]
 	return bz, true
 }
 
+// GetLastCompletedUpgrade returns the last applied upgrade name and height.
+func (k Keeper) GetLastCompletedUpgrade(ctx sdk.Context) (string, int64) {
```

Comment:
side note: we are polluting the exposed functions only because we prioritize external test (with `_test` package name)

Reply:
We are using this function in `BeginBlocker` too.